src/modules/owner/owner.service.ts

```diff
@@ -0,0 +1,31 @@
+import { Injectable } from '@nestjs/common';
+import { PrismaService } from 'src/common/prisma/prisma.service';
+import { Response } from 'express';
+import { UserRoleType } from '../user/enums/user-role.enum';
+import { AuthService } from '../auth/auth.service';
+import { CreateUserDto } from '../auth/dto/create-user.dto';
+import { AuthResponse } from '../auth/interfaces/auth-response.interface';
+
+@Injectable()
+export class OwnerService {
+  constructor(
+    private prismaService: PrismaService,
+    private authService: AuthService,
+  ) {}
+
+  public async registration(
+    userData: CreateUserDto,
+    res: Response,
+  ): Promise<AuthResponse> {
+    const user = await this.authService.registration(userData, res);
+    const role = await this.prismaService.role.upsert({
+      where: { name: UserRoleType.PROPERTY_OWNER },
+      update: {},
+      create: { name: UserRoleType.PROPERTY_OWNER },
+    });
```

Comment:
Maybe just use findUnique? I don't think that there is a point  to check whether it exists or not

Reply:
upsert is create if does not exist, if exist just update, if data is the same it is skipping it. High performance, however, could be changed later

Reply:
Higher performance than findUnique?

Reply:
Actually, this upset will be executed only once, any other calls of the function will not create propertyOwner role. So, it is better to use findUnique. The basic roles may be created in seed I think